tests/http_integration.rs

```diff
@@ -959,16 +959,6 @@ mod http_integration {
 			assert!(res.is_ok(), "upgrade err: {}", res.unwrap_err());
 		}
 
-		// Test nul character
-		{
-			let res =
-				client.post(url).body("parse::email::user('\\u0000@example.com')").send().await?;
-			assert_eq!(res.status(), 400);
-
-			let body = res.text().await?;
-			assert!(body.contains("contained NUL byte"), "body: {body}");
-		}
-
```

Comment:
Could we find another query that would trigger a value conversion error, so that the error handling is still tested?

Reply:
You shouldn't be able to from SurQL which is why I implemented this fix. Null bytes in strings can cause all kinds of issues if they make it to the KV store. 

I can trigger the error by calling the parse::email::user method from rust so I will do that.